src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import duke.command.*;
+import duke.exception.DukeException;
+
+/**
+ * Translate text to command.
+ */
+class Parser {
+    private static final int TOKEN_NUM = 2;
+    private TaskList tasks;
+    private Storage storage;
+
+    /**
+     * Constructor of a parse
+     * @param tasks tasks
+     */
+    public Parser(TaskList tasks, Storage storage) {
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * From input to determine which type of command to generate
+     * @param input scanner's result
+     * @return the generated command for execute
+     * @throws DukeException wrong usage
```

Comment:
In general it seems that it would be better if you add a blank line before the `@param` block in all your JavaDoc. In addition, it would be more faithful to coding standard if you phrase the description like `Determines which type of command to generate from input`, which foregrounds the action verb (in third person singular).